index.js

```diff
@@ -28,6 +28,13 @@ function File(file) {
   this.contents = file.contents || null;
 
   this._isVinyl = true;
+
+  // Set custom properties
+  Object.keys(file).forEach(function(key) {
+    if (this.constructor.isCustomProp(key)) {
+      this[key] = file[key];
+    }
+  }, this);
```

Comment:
I prefer to avoid this binding.  I know `self` is messier but I'm always nervous about bind performance